src/main/java/org/jenkinsci/plugins/mesos/MesosRetentionStrategy.java

```diff
@@ -94,6 +96,22 @@ private long checkInternal(MesosComputer c) {
         }
       }
     }
+
+    // Terminate the computer if it is exists for longer than
+    // 'maximumTimeToLive'.
+    final long timeLivedInMilliseconds =
+            DateTimeUtils.currentTimeMillis() - c.getConnectTime();
+
+    if (c.isOnline()) {
+
+      if (timeLivedInMilliseconds > MINUTES.toMillis(maximumTimeToLive)) {
+        LOGGER.info("Disconnecting computer greater maximum TTL " + c.getName());
+
```

Comment:
Let me ask a question. Can slave be removed without setting pendingDelete, is anything else of safe clener here except for MesosCleanupThread ?
(I look into these piece of  codes recently, suffering from slave connectivety issue between plugin and mesos/zookeeper in my test environment.)